File: docs/Software/POO/EP/ep_python/S1000.py
from Drone import Drone
class S1000(Drone):
    def __init__(self, nome, bateria, posicao):
        super().__init__(nome, bateria, posicao)

    def mapear(self):
        if self._decolado:
            if self._bateria.usar(10):
                print("Mapeando...")
                print("_____________________(oooo)__")
                print("___o-o______\O/________||____")
                print("_____________|_________||____")
                print("_____________/\________||____")
                print("_____________________________")
                print("Mapeamento feito!\n")
                return True
            return False
        else:
            print("Drone nao esta decolado!")
            return False

File: docs/old/Software/POO/EP/ep_cpp/S1000.cpp
#include "S1000.hpp"
#include <iostream>
using namespace std;

S1000::S1000(string nome, Bateria* bateria, double posicao) : Drone(nome, bateria, posicao){
}

S1000::~S1000(){
    cout << "S1000 destruido" << endl;
}

int S1000::mapear(){
    if(decolado == false){
        cout << "Drone esta no chao" << endl;
        return 1;
    }
    if(bateria->getCarga() < 10){
        cout << "sem carga" << endl;
        return 1;
    }
    else{
        bateria->usar(10);
        cout << "Mapeando..." << endl;
        cout << "_________________________" << endl;
        cout << "___o-o_______________/\\__" << endl;
        cout << "_____________________||__" << endl;
        cout << "___________\\O/_______||__" << endl;
        cout << "____________|________||__" << endl;
        cout << "____________/\\_______||__" << endl;
        cout << "_____________________||__" << endl;
        cout << "Mapeamento realizado" << endl;
        return 0;
    }
}
